script.js

```diff
@@ -1,79 +1,92 @@
 });
 
-prevBtn.addEventListener('click', () => {
-    playBackward();
+prevBtn.addEventListener("click", () => {
+  playBackward();
 });
 
-nextBtn.addEventListener('click', () => {
-    playForward();
+nextBtn.addEventListener("click", () => {
+  playForward();
 });
 
- function stopMusic() {
-    if (isPlaying() == false) {
-        return;
-    }
-    playBtn.querySelector('i').classList.remove('fa-pause');
-    playBtn.querySelector('i').classList.add('fa-play');
-    music.pause();
- }
-
- function startMusic() {
-    if (isPlaying() == true) {
-        return;
-    }
-    playBtn.querySelector('i').classList.remove('fa-play');
-    playBtn.querySelector('i').classList.add('fa-pause');
-    music.play();
- }
-
- function isPlaying() {
-    return playBtn.querySelector('i').classList.contains('fa-pause');
- }
-
- function playForward() {
-    stopMusic();
-    index = (index + 1) % songList.children.length;
-    updateMusic(index);
-    startMusic();
- }
-
- function playBackward() {
-    stopMusic();
-    index = (index - 1);
-    if (index < 0) index = songList.children.length - 1;
-    updateMusic(index);
-    startMusic();
- }
-
- function updateMusic(index) {
-    if (currentSongUl !== undefined)
-        currentSongUl.classList.remove('playing');
-
-    currentSongUl = songList.children[index];
-    currentSongName = currentSongUl.innerText;
-    music = new Audio(`./audio/${currentSongName}.mp3`);
-
-    currentSongUl.classList.add('playing');
- }
-
-
-
- 
-
- 
\ No newline at end of file
+function stopMusic() {
+  if (isPlaying() == false) {
+    return;
+  }
+  playBtn.querySelector("i").classList.remove("fa-pause");
+  playBtn.querySelector("i").classList.add("fa-play");
+  music.pause();
+}
+
+function startMusic() {
+  if (isPlaying() == true) {
+    return;
+  }
+  playBtn.querySelector("i").classList.remove("fa-play");
+  playBtn.querySelector("i").classList.add("fa-pause");
+  music.ontimeupdate = function (e) {
+    progressed.style.width = (music.currentTime * 100) / music.duration + "%";
+  };
+  progressBar.onclick = function (e) {
+    music.currentTime = (e.offsetX / progressBar.offsetWidth) * music.duration;
+  };
+  music.play();
+}
+
+function isPlaying() {
+  return playBtn.querySelector("i").classList.contains("fa-pause");
+}
+
+function playForward() {
+  stopMusic();
+  index = (index + 1) % songList.children.length;
+  updateMusic(index);
+  startMusic();
+}
+
+function playBackward() {
+  stopMusic();
+  index = index - 1;
+  if (index < 0) index = songList.children.length - 1;
+  updateMusic(index);
+  startMusic();
+}
+
+function updateMusic(index) {
+  if (currentSongUl !== undefined) currentSongUl.classList.remove("playing");
+
+  currentSongUl = songList.children[index];
+  currentSongName = currentSongUl.innerText;
+  music = new Audio(`./audio/${currentSongName}.mp3`);
+
+  currentSongUl.classList.add("playing");
+}
+
+function selectMusic() {
+  stopMusic();
+
+  let li = event.target.closest("li");
```

Comment:
"event" is deprecated, we should avoid it. right ?
Also we can pass this event by "onclick="selectMusic(this)"" from html and use it in js.